### DISCUSSION TITLE
Compute shaders running in browser

### DISCUSSION THREAD

**Discussion by bmahlbrand** [gfx-rs/wgpu#2638] [Q&A]
Sorry if this is a stupid question but I was curious if there's a clear answer on whether I can expect compute pipelines to more or less work as expected on web at this time.

i.e. should this example work?

https://github.com/gfx-rs/wgpu/blob/master/wgpu/examples/hello-compute/README.md

**Comment by cwfitzgerald** [comment]
Yes it should, baring issues from mismatched versions of wgsl or WebGPU.

**Reply by bmahlbrand** [reply]
Ok, maybe documenting this will help someone else, so I installed the latest version of rust (on Win10), and followed instructions here (running latest Firefox nightly as of 5/3/22):

`https://github.com/gfx-rs/wgpu/wiki/Running-on-the-Web-with-WebGPU-and-WebGL`

installed `wasm32-unknown-unknown` as suggested when building the example for web failed ala `rustup target add wasm32-unknown-unknown`

but I'm getting:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: failed to read `target/wasm-examples-target/wasm32-unknown-unknown\debug\examples\boids.wasm`

Caused by:
    0: failed to read `target/wasm-examples-target/wasm32-unknown-unknown\debug\examples\boids.wasm`
    1: The system cannot find the path specified. (os error 3)', C:\Users\ben\.cargo\registry\src\github.com-1ecc6299db9ec823\cargo-run-wasm-0.1.0\src\lib.rs:135:10
```

What step did I miss here? Same error for all the ones I tried, `hello`, `hello-compute`, `boids` etc

[update] perhaps the answer lies within https://gfx-rs.github.io/2020/04/21/wgpu-web.html
